epitome/models.py

```diff
@@ -746,3 +751,148 @@ def create_model(self, **kwargs):
+                 label_cell_types=self.test_celltypes,   # used for labels. Should be all for train/eval and subset for test
+                 eval_cell_types=self.eval_cell_types,   # used for rotating features. Should be all - test for train/eval
+                 matrix=self.dataset.matrix,
+                 targetmap=self.dataset.targetmap,
+                 cellmap=self.dataset.cellmap,
+                 radii = self.radii,
+                 mode = Dataset.RUNTIME,
+                 similarity_matrix = matrix,
+                 similarity_targets = ['DNase'],
+                 indices = indices,
+                 continuous=True,
+                 keep_lbl_mask=False,
+                 return_feature_names=False)
+
+        to_stack = load_data(data=self.dataset.get_data(Dataset.ALL),
+                 label_cell_types=self.test_celltypes,   # used for labels. Should be all for train/eval and subset for test
+                 eval_cell_types=self.eval_cell_types,   # used for rotating features. Should be all - test for train/eval
+                 matrix=self.dataset.matrix,
+                 targetmap=self.dataset.targetmap,
+                 cellmap=self.dataset.cellmap,
+                 radii = self.radii,
+                 mode = Dataset.RUNTIME,
+                 similarity_matrix = matrix,
+                 similarity_targets = ['DNase'],
+                 indices = indices,
+                 continuous=True,
+                 return_feature_names=True)
+
+        gen_to_list = list(gen())
+        to_stack = list(to_stack())
+        gen_to_list = np.array(gen_to_list)
+
+        # reshape to n_regions [from regions] x nassays [acc dim 1] x n_samples
+        radii = self.radii
+
+        stacked = np.stack([to_stack] * accessibility_peak_matrix.shape[0], axis=0)
+        names = stacked[:, :, 1]
+        to_stack = stacked[:, :, 0]
+        to_stack = np.expand_dims(to_stack, axis=-1)
+
+        same_size = accessibility_peak_matrix.shape[1] == len(conversionObject.joined.idx_base)
+
+        if not same_size:
+            added_indices = []
+            old_idx, counter, old_i = 0, 0, 0
+            indices_to_merge = []
+            for ctr, (i, i_base) in enumerate(zip(conversionObject.joined.idx, conversionObject.joined.idx_base)):
+                if i_base == -1:
+                    continue
+                if i != old_i:
+                    indices_to_merge.append((old_idx, counter))
+                    old_idx = counter
+                added_indices.append(accessibility_peak_matrix[:, i])
+                counter += 1
+                old_i = i
+            indices_to_merge.append((old_idx, len(conversionObject.joined.idx)))
+            
+            a = np.stack(added_indices)
+        else:
+            a = np.transpose(accessibility_peak_matrix, axes=[1, 0])
+        
+        a = a[:, None, :]
+        
+        out = compute_casv(gen_to_list, a, radii)
```

Comment:
Upon looking at the new score_matrix function, I believe it is exhibiting incorrect behavior, and compute_casv is not the issue (although it should be able to handle a division by 0 error). The issue is that you are passing in a 4 x 1 x 2 matrix for m2, and similarly shaped for m1. However, the region dimensions should be the full size of the dataset (in the test case, 2100 regions). This correct behavior can be seen in the original score_matrix function, where you pass in the entire matrices, then use the indices parameter (now in compute_casv) to specify the indices you want to compute on